.github/workflows/asan_ubsan_build.yml

```diff
@@ -0,0 +1,47 @@
+
+name: Build with ASAN and UBSAN
+
+on:
+  workflow_dispatch:
+  pull_request: 
+    branches: [trunk]
+    types: [opened, ready_for_review]
+
+env:
+  LDMX_DOCKER_TAG: ldmx/dev:latest
+
+jobs:
+  build-with-asan-ubsan:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout ldmx-sw
+      uses: actions/checkout@v4
+      with:
+        submodules: 'recursive'
+        fetch-depth: 0
+
+    - name: Install just
+      uses: extractions/setup-just@v2
+      with:
+          just-version: 1.26.0
+ 
+    - name: Compile and Install ldmx-sw
+      run: |
+        just install-denv init configure-asan-ubsan 
+        just build
+      shell: bash
+
+    - name: Run ecal_pn config
+      run: |
+        just setenv LDMX_NUM_EVENTS=1500
+        just setenv LDMX_RUN_NUMBER=1
+        export LDMX_NUM_EVENTS=15
+        export LDMX_RUN_NUMBER=1
```

Comment:
```suggestion
```

These lines don't get passed into the containerized environment since `just init` tells `denv` to not share host environment variables by default. And when setting an environment variable, `denv` overrides the host's value (you can see that fire ran for 1500 events rather than just 15).